### PR TITLE
call db functions on background thread in ConversationsViewModel

### PR DESCRIPTION
**Pull request by connyduck**
really nice that they added `Completable` and `Single` support directly to Room in 2.1.0

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsViewModel.kt [hunk 1]
```diff
@@ -67,6 +71,8 @@ class ConversationsViewModel  @Inject constructor(
                     lastStatus = conversation.lastStatus.copy(expanded = expanded)
             )
             database.conversationDao().insert(newConversation)
+                    .subscribe()
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/components/conversation/ConversationsViewModel.kt]: I don't think it makes much sense to add them to disposables in all 3 cases. We don't want to cancel these actions when the screen is closed, right? We still want to update DB or something.

File: app/build.gradle [hunk 2]
```diff
@@ -112,9 +112,9 @@ dependencies {
     // architecture components
     implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
     //room
-    implementation 'androidx.room:room-runtime:2.0.0'
-    kapt 'androidx.room:room-compiler:2.0.0'
-    implementation 'android.arch.persistence.room:rxjava2:1.1.1'
+    implementation 'androidx.room:room-runtime:2.1.0-alpha04'
```
Comment (charlag) [inline app/build.gradle]: I think that you can return `Single` from the inception:
https://medium.com/androiddevelopers/room-rxjava-acb0cd4f3757
(post from June 207)
So no need for alpha, really

File: app/src/main/java/com/keylesspalace/tusky/db/ConversationsDao.kt [hunk 3]
```diff
@@ -18,17 +18,18 @@ package com.keylesspalace.tusky.db
...
+import io.reactivex.Completable
 
 @Dao
 interface ConversationsDao {
     @Insert(onConflict = OnConflictStrategy.REPLACE)
     fun insert(conversations: List<ConversationEntity>)
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    fun insert(conversation: ConversationEntity)
+    fun insert(conversation: ConversationEntity): Completable
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/db/ConversationsDao.kt]: If we don't want to cancel them (I think we don't) we could as well leave them as blocking ones.

File: app/build.gradle (hunk 2 above)
Comment (connyduck) [reply app/build.gradle]: nope, in the article you linked: 

> Later edit 2: starting with Room 2.1.0-alpha01, DAO methods annotated with @Insert, @Delete or @Update support Rx return types Completable, Single<T> and Maybe<T>.

Comment (charlag) [reply app/build.gradle]: Ah, sorry, I misunderstood, this are mutation functions! They've been always blocking and now you can subscribe on another thread, that's really great.

**Pull request merged**